crates/ouroboros/src/validator/mod.rs

```diff
@@ -1,6 +1,7 @@
+use serde::{Deserialize, Serialize};
 use thiserror::Error;
 
-#[derive(Error, Debug)]
+#[derive(Error, Debug, PartialEq, Serialize, Deserialize)]
```

Comment:
What requires those new derive attributes?

Reply:
the snapshots we record on disk, precisely this line: https://github.com/pragma-org/amaru/blob/ed049ed26036de350b20edc683ea0c9558a10e76/crates/ouroboros/src/consensus/test.rs#L117

Reply:
Ok, looks a bit weird to have to change Error definitions for a specific test. Maybe we could avoid storing errors somehow?

Reply:
I am not a great fan of snapshot tests so perhaps we could do that. In general however, I don't see much of a problem to have an error that's serialisable. Rather the opposite actually.